### PR TITLE
DIGITAL-3247 Add Redis for Post Data

### PR DESCRIPTION
**Pull request by tirish**
Add Redis data store for post data. Post data will continue to be stored in a flat file, the same as it is does now, but now there is the ability to read/clear post data using a key.

The key will be the email field value on the form, and if none is present, will be set to `invalid`, which can still be used for checking.

To check post data use the same post data endpoint but include: `?key=<email>`, where `<email>` is the email value from the form post. If you click the button on the post data page that loads, it will only clear that entry from redis, the flat file will be unaffected. 

Clicking the `Clear Post Data` button on the regular post data page will clear both redis and the flat file.

The environment variable `REDISCLOUD_URL` should be set on Heroku. Also, `REDIS_EXPIRE_TIME` may be set on Heroku, the default time before a entry in redis expires is 30min. If set to 0 or lower, it will never expire.

### CODE REVIEW

File: lib/data/index.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+
+function setData(key,data){
+  console.log('setting data');
+  key = formatKey(key);
+
+  return romis.set(key,JSON.stringify(data))
+    .then(function(){
+      console.log('setting data expiration: ' +expireTime +' seconds');
+      if(expireTime > 0) {
```
Comment (kthayer424) [inline lib/data/index.js]: Do you need this if?

File: lib/data/index.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        return romis.expire(key, expireTime);
+      } else {
+        console.log('NOT setting data expiration');
+        return resolve(data);
```
Comment (kthayer424) [inline lib/data/index.js]: `BPromise.resolve(data);`

File: lib/data/index.js (hunk 1 above)
Comment (tirish) [inline lib/data/index.js]: So if you don't want an expiration you set the environment variable to 0 or less.

Comment (kthayer424) [inline lib/data/index.js]: It's alright to call expire with 0 or a negative number.

Comment (tirish) [inline lib/data/index.js]: Oh, will that behave the same way?

Comment (kthayer424) [inline lib/data/index.js]: Yeah, the data just won't persist.

**Pull request merged**